pkg/testutil/nerdtest/ambient.go

```diff
@@ -0,0 +1,31 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package nerdtest
+
+import "github.com/containerd/nerdctl/v2/pkg/testutil"
+
+func environmentHasIPv6() bool {
+	return testutil.GetEnableIPv6()
+}
+
+func environmentHasKubernetes() bool {
+	return testutil.GetEnableKubernetes()
+}
+
+func environmentIsForFlaky() bool {
+	return testutil.GetFlakyEnvironment()
+}
```

Comment:
Why not just call testutil functions directly

Reply:
Currently we have to bridge between the legacy tests, and the code handling flag parsing, and the proposed new framework, which unfortunately implies a certain level of duplication.
I will clean that up once all tests are migrated.